hazelcast/src/main/java/com/hazelcast/jet/impl/LightMasterContext.java

```diff
@@ -110,6 +110,8 @@ public LightMasterContext(
             logFine(logger, "Light job %s will run on a subset of members: %d out of %d members with version %s",
                     idToString(jobId), members.size(), membersView.size(), coordinatorVersion);
         }
+
+        dag.lowerDownParallelism(nodeEngine.getConfig().getJetConfig().getCooperativeThreadCount());
```

Comment:
This certainly must not happen here. The DAG api is supposed to be low level - what parallelism and edge you specify, you'll get it. It should be in the `CreateDagVisitor`.

Reply:
I've changed it, but look is it a proper way. The simplest way is to execute ```lowerDownParallelism``` when DAG is completed. If you prefer to change parallelismy during building DAG in ```CreateDagVisitor``` let me know. It is doable.